### ISSUE TITLE
Ability to enter IP of Printer(s)

### ISSUE THREAD

**Issue by TheCableGuy99** [bambulab/BambuStudio#2232]
**Is your feature request related to a problem? Please describe.**
Yes, I've started to use LAN mode now video is supported. But the printers reside on a separate IoT subnet where scanning is not allowed and therefore printers cannot be auto-discovered.

**Describe the solution you'd like**
The ability to enter the IP address of the printer and it allows it to be a different subnet. For example, if my PC is on 192.168.1.10 and one of the printers is on 192.168.2.10 then I can enter rhis and it stores it until I decide to edit it. This must support multiple printers.

**Describe alternatives you've considered**
Putting my PC and printer on the same subnet and allowing scanning is the only way to get LAN mode working that I can see.

**Additional context**
I think the above is clear enough but please let me know if you need any further clarification.

**Comment by Znubbis** [comment]
+1, if they want to sell printer to corporations this is required to be possible.

**Comment by Alex-Schaefer** [comment]
This feature is an absolute necessity, why is this not implemented?

Bambu Studio can't find the printer in LAN only mode, because different network, but using FTP to connect works. So it's not a network problem, it's a Bambu Studio Problem. I suppose being able to input the printer IP it should work in Bambu Studio, too.

Can't use the printer like that in the company.

**Comment by timothyzalusky** [comment]
Adding a +1 to try and get some more visibility to this feature.  Expectation would be that this should simplify users like me who have a printer at home and Bambu Printers at work and would like to keep them separate.

**Comment by Technology89** [comment]
> There is an ugly workaround: Send a fake SSDP discovery packet.
> 
> Either use this Linux only script from #702: https://github.com/gashton/bambustudio_tools/blob/master/bambudiscovery.sh Or my Python implementation (tested on Windows 10): https://gist.github.com/Alex-Schaefer/72a9e2491a42da2ef99fb87601955cc3
> 
> Nevertheless, this issue should be addressed by Bambu

Thanks for the Python Script. Is there some specific thing i need to put in for the P1S? Tried "3DPrinter-P1S" but it still shows "Model: Unknown" in Update Page and can't access the camera.

Btw +1 for this feature.

**Comment by OndrejLycka** [comment]
I, too, miss that option a lot.

**Comment by wilk-polarny** [comment]
Yes, such an option is essential. Especially IF printers are on different subnets, the regular UDP announcement will not work. I have a separate WiFi and network for all IoT stuff and would like to keep things isolated.

The workarounds posted above do work, but I'll have to send fake broadcasts each time I open up bambu studio...

Please, add an option to add printer IP addresses and REMEMBER them in Bambu Studio!

**Comment by hoegge** [comment]
It is quite crazy you cannot add a printer by IP - it means a lot of user simply cannot connect to it and means you can return the printer and get a refund, since it does not work.

**Comment by ColColonCleaner** [comment]
We really need the ability to add printers by IP. I cannot use an X1C on my network because of this problem.

The scripted workarounds are not working for me either. Not sure if that was blocked in a recent BambuStudio update, which would be very bad.

**Comment by timothyzalusky** [comment]
@walterwongbbl I would expect 10k users actually want this implemented (or more).  This needs to be a priority.  Can you give us a rough timeline for implementation?

**Comment by SaltWei** [comment]
This will be supported in next studio version

**Comment by ColColonCleaner** [comment]
> This will be supported in next studio version

Excellent news. Looking forward to the update. Cheers.

**Comment by timothyzalusky** [comment]
This will close out almost 10 different issue items, excited to see it.

**Comment by timothyzalusky** [comment]
@SaltWei Just saw a new release of studio just dropped, but it looks like this feature was left out.  Any timeline for the implementation of this feature?

**Comment by alebre-lithoz** [comment]
I know its a bit early since the new new release is only 40 minutes old but these feature wasn't implemented as far as I see.

**Comment by wilk-polarny** [comment]
> I know its a bit early since the new new release is only 40 minutes old but these feature wasn't implemented as far as I see.

They said `next studio version`, but not which one ;)

**Comment by SaltWei** [comment]
That is a queue jumping release for some bug fix. The version for this issue is next big release 1.10 which may come 1.5 month later.

**Comment by ColColonCleaner** [comment]
@SaltWei thank you for confirming this fix is coming and the specific version it's coming in.
Understandable it would be a larger release since both studio and the network plugin require updates for this.

**Comment by meisnick** [comment]
This is still absolutely unacceptable for the X1E a "Enterprise" sale product for institutions to have to work around this issue. This has been open for Over a year please provide a timeline for this.

**Comment by bradich** [comment]
Looks like this is coming in the next release. It's in the public beta of Studio now. https://github.com/bambulab/BambuStudio/releases/tag/v01.10.00.74

**Comment by giggls** [comment]
> Looks like this is coming in the next release. It's in the public beta of Studio now. https://github.com/bambulab/BambuStudio/releases/tag/v01.10.00.74

Which does not work. See previous comment of @meisnick

**Comment by ColColonCleaner** [comment]
> "The device does not support using ip and Access Code for connection"

Wait... How is it possible that the printer "does not support" connection by IP and Access Code? That's how it already works if you spoof the SSDP data, so the only change is making studio be able to manually enter the IP.

Unless there is currently no way to get the serial number from the printer without the SSDP data? I know when integrating with HA it requires the serial number to be known ahead of time. Did they need a firmware update just to get the serial number? For anyone wanting to enter an IP manually, they would also have the serial number on hand, I'm fine with entering that myself too.

**Comment by giggls** [comment]
Whatever they did the functionality seems to be there in public beta but does not work.

**Comment by mwdle** [comment]
This functionality is also not working for me on the beta. My issue is described on [this thread](https://github.com/bambulab/BambuStudio/issues/702#issuecomment-2424388727)

**Comment by AndreasSchwalb** [comment]
Maybe anybody is interested in:
I wrote a small relay which take the SSDP messages the printer send over the network and send it to the machine where BambuStudio is running on.
For my case it is a nice workaround. Just spin up the container and BambuStudio is happy.

https://github.com/AndreasSchwalb/babulabRelay

**Comment by RX14** [comment]
You can also perform redirection of the bambu discovery between LANs in nftables, with the following rule:

```
iifname "<input dev>" ip daddr 255.255.255.255 udp dport 2021 dup to 255.255.255.255 device "<output dev>" comment "bambu service discovery"
```

where the input and output device can be different physical network interfaces, or vlan interfaces etc. This rule needs to go in a prerouting hook chain, and only works in tables of type `ip` *not* `inet`.

**Comment by JanikMie** [comment]
Is there any way by now to get this working without rerouting SSDP messages or sth?
For our Enterprise Network i was not able to get it working. In the Same Lan it works... different Lan the Printer refuses the connection. We already have multiple BambuLab Printers but for testing i am using a P1P.

**Comment by JanikMie** [comment]
> [@JanikMie](https://github.com/JanikMie) Try Orca Slicer if that's an option for you. They have a workaround for the broken network plugin that allows you to add the printer by IP (and it actually works across subnets).

Thanks i will try that.
